python_modules/dagster/dagster_tests/components_tests/resolution_tests/test_resolved.py

```diff
@@ -185,3 +185,16 @@ def __init__(
 
     with pytest.raises(ResolutionException, match="positional only parameter"):
         PosOnly.resolve_from_yaml("")
+
+
+def test_nested_not_resolvable():
+    @dataclass
+    class Child:
+        name: str
```

Comment:
the other option would be to just derive schemas for any types we hit that qualify that are children of a `Resolvable` type instead of having them be explicitly opted-in